src/main/java/de/hysky/skyblocker/skyblock/dungeon/puzzle/waterboard/WaterboardPreviewer.java

```diff
@@ -0,0 +1,226 @@
+package de.hysky.skyblocker.skyblock.dungeon.puzzle.waterboard;
+
+import de.hysky.skyblocker.annotations.Init;
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.skyblock.dungeon.puzzle.DungeonPuzzle;
+import de.hysky.skyblocker.skyblock.dungeon.secrets.DungeonManager;
+import de.hysky.skyblocker.skyblock.dungeon.secrets.Room;
+import de.hysky.skyblocker.utils.ColorUtils;
+import de.hysky.skyblocker.utils.render.RenderHelper;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderContext;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.util.Pair;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.BlockView;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static de.hysky.skyblocker.skyblock.dungeon.puzzle.waterboard.Waterboard.*;
+
+public class WaterboardPreviewer extends DungeonPuzzle {
+	private static final Logger LOGGER = LoggerFactory.getLogger(WaterboardPreviewer.class);
+	public static final WaterboardPreviewer INSTANCE = new WaterboardPreviewer();
+
+	private LeverType prospective;
+	private ClientWorld world;
+	private Room room;
+	private ClientPlayerEntity player;
+
+	private WaterboardPreviewer() {
+		super("waterboard", "water-puzzle");
+	}
+
+	@Init
+	public static void init() {
+	}
+
+	@Override
+	public void tick(MinecraftClient client) {
+	}
+
+	@Override
+	public void render(WorldRenderContext context) {
+		world = MinecraftClient.getInstance().world;
+		room = DungeonManager.getCurrentRoom();
+		player = MinecraftClient.getInstance().player;
+		if (!shouldSolve() || world == null || player == null || !DungeonManager.isCurrentRoomMatched()) {
+			return;
+		}
+
+		try {
+			findProspective();
+			renderWaterPath(context);
+			renderProspectiveChanges(context);
+		} catch (Exception e) {
+			LOGGER.error("[Skyblocker Waterboard] Error while rendering previews", e);
+		}
+	}
+
+	private void renderWaterPath(WorldRenderContext context) {
+		if (!SkyblockerConfigManager.get().dungeons.puzzleSolvers.previewWaterPath) {
+			return;
+		}
+
+		// Calculate and render path of water through the board
+		List<Pair<BlockPos, BlockPos>> waterPath = new ArrayList<>();
+		waterPath.add(new Pair<>(WATER_ENTRANCE_POSITION.up(5), WATER_ENTRANCE_POSITION.up(3)));
+		findWaterPathVertical(WATER_ENTRANCE_POSITION.up(3), waterPath);
```

Comment:
`findWaterPathVertical` and `findWaterPathHorizontal` accounts for `prospective`. I think it would be a good idea to comment details like that.